STABILITY.md

```diff
@@ -0,0 +1,123 @@
+# Stability
+
+Even though this project adheres to [semver], this guide aims to explain how semantic versioning is used exactly and how we aim to provide stability within an
+ever-changing codebase.
+
+Please note that this guide isn't stable itself and may be adjusted to fit changes in requirements or new ways are discovered.
+
+## Terminology
+
+* _dependent crate_
+  - A crate which depends on a crate in this workspace directly.
+* _downstream crate_
+  - A crate which directly or indirectly depends on a crate in this workspace.
+* _workspace crate_
+  - A crate which is a member of this workspace and hence is stored in this repository
+* _breaking change_
+  - A change in code that requires a `dependent crate` to adjust their code to fix compile errors.
+* _release_
+  - A new version of a crate is published to crates.io 
+
+## Tiers
+
+The project uses three stability tiers for all of its crates, and all crates use [semver] for their version numbers.
+Tiers differ primarily in the time between breaking changes, which always have to be announced with `PRs` as per
+our [collaboration guide].
+
+The following schematic helps to visualize what follows.
+
+```
+                Release Software v1.X              
+    Stability Tier 1 ═════════════════════════════╗
+    ║                                             ║
+    ║    gixp─────────────┐ gix──────────────┐    ║
+    ║    │  plumbing app  │ │  porcelain app │    ║
+    ║    └────────────────┘ └────────────────┘    ║
+    ║             │                  │            ║
+    ║             ▼                  ▼            ║
+    ║    gitoxide-core───────────────────────┐    ║
+    ║    │        application functionality  │    ║
+    ║    └───────────────────────────────────┘    ║
+    ║                      │                      ║
+    ║                      ▼                      ║
+    ║    git-repository──────────────────────┐    ║
+    ║    │                application crate  │    ║
+    ║    └───────────────────────────────────┘    ║
+    ║                      │                      ║
+    ╚═════════════════════════════════════════════╝
+                           │                       
+    Stability Tier 2 ─────────────────────────────┐
+    │                      ▼                      │
+    │    plumbing crates─────────────────────┐    │
+    │    │ ┌───────────┐       ┌───────────┐ │    │
+    │    │ │  git-odb  │       │  git-ref  │ │    │
+    │    │ └───────────┘       └───────────┘ │    │
+    │    │ ┌───────────┐       ┌───────────┐ │    │
+    │    │ │git-config │       │ git-pack  │ │    │
```

Comment:
`git-config` is still pre-1.0, which would be a tier 3 stability.

Reply:
How can it be made clearer that this is an example for the structure of crates and their maximum stability tier? 

Looking at it again I do realize that `git-repository` plans to provide direct struct field access to something from `git-config`, so that would make it ST1.

It's probably better to hide these types or to commit to more stability once it goes 1.0.